diff -Naur libdnet-1.7-p/test/check/Makefile.am libdnet-1.7/test/check/Makefile.am
--- libdnet-1.7-p/test/check/Makefile.am	Tue Apr  1 13:48:30 2003
+++ libdnet-1.7/test/check/Makefile.am	Tue Apr  1 13:59:24 2003
@@ -23,4 +23,4 @@
 check_rand_SOURCES = check_rand.c
 check_route_SOURCES = check_route.c
 
-LDADD = $(top_builddir)/src/libdnet.a @CHECKLIB@
+LDADD = $(top_builddir)/src/libdnet.la @CHECKLIB@
diff -Naur libdnet-1.7-p/test/dnet/Makefile.am libdnet-1.7/test/dnet/Makefile.am
--- libdnet-1.7-p/test/dnet/Makefile.am	Mon Jan 20 07:44:35 2003
+++ libdnet-1.7/test/dnet/Makefile.am	Tue Apr  1 13:59:14 2003
@@ -11,7 +11,7 @@
 
 EXTRA_DIST = $(man_MANS)
 
-LDADD = $(top_builddir)/src/libdnet.a
+LDADD = $(top_builddir)/src/libdnet.la
 
 dnet.8.txt: dnet.8
 	groff -t -e -mdoc -Tascii dnet.8 | col -bx > $@
diff -Naur libdnet-1.7/configure.in libdnet-1.7-p/configure.in
--- libdnet-1.7/configure.in	Fri Mar 21 06:46:37 2003
+++ libdnet-1.7-p/configure.in	Tue Apr  1 14:01:52 2003
@@ -23,7 +23,7 @@
 dnl Checks for programs.
 AC_PROG_CC
 AC_PROG_INSTALL
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 dnl XXX - stupid IRIX cpp
 if test -r /usr/include/sgidefs.h ; then
 	CPPFLAGS="$CPPFLAGS -D__sgi"
diff -Naur libdnet-1.7/src/Makefile.am libdnet-1.7-p/src/Makefile.am
--- libdnet-1.7/src/Makefile.am	Tue Apr  1 14:02:39 2003
+++ libdnet-1.7-p/src/Makefile.am	Tue Apr  1 14:03:35 2003
@@ -2,8 +2,8 @@
 
 include $(top_srcdir)/Makefile.am.common
 
-lib_LIBRARIES = libdnet.a
+lib_LTLIBRARIES = libdnet.la
 
-libdnet_a_SOURCES = addr-util.c addr.c blob.c ip-util.c ip6.c rand.c
+libdnet_la_SOURCES = addr-util.c addr.c blob.c ip-util.c ip6.c rand.c
 
-libdnet_a_LIBADD = @LIBOBJS@
+libdnet_la_LIBADD = @LIBOBJS@
